internal/reminder/reminder.go

```diff
@@ -74,21 +78,52 @@ func (r *reminder) Start(ctx context.Context) error {
 		return errors.New("reminder stopped, cannot start again")
 	default:
 	}
+	defer r.Stop()
 
 	interval := r.cfg.RecurrenceConfig.Interval
 	if interval <= 0 {
 		return fmt.Errorf("invalid interval: %s", r.cfg.RecurrenceConfig.Interval)
 	}
 
+	metricsServerDone := make(chan struct{})
+
+	if r.cfg.MetricsConfig.Enabled {
+		metricsProviderReady := make(chan struct{})
+
+		go func() {
+			if err := r.startMetricServer(ctx, metricsProviderReady); err != nil {
+				logger.Err(err).Msg("failed to start metrics server")
+			}
+			close(metricsServerDone)
+		}()
```

Comment:
It feels weird to me to end up needing to call `startMetricServer` in another thread and then deal with that, given that `startMetricServer` _also_ sets up a background thread (and then waits for it).

We end up with:

```
main
 \--> startMetricServer
          \--> http.ListenAndServe
```

When it feels like we should have:

```
main
\--> http.ListenAndServe
```

Possibly with a "closer" callback.

I only mention this because `Join`-ing goroutines is a bunch of code that distracts from the rest of what's going on, so I tend to prefer to minimize how often I need to block on a goroutine exiting.

Reply:
Yeah, that's a good point. It was confusing especially with the part of getting the otel meter provider ready and set. I modified the implementation to be like:

```
main
\--> http.ListenAndServer
\--> watch for stopping the metrics server
```

It can't be done in a single go routine, but I think this solution is better than what was there previously.